extension/src/popup/Router.tsx

```diff
@@ -96,6 +108,10 @@ const PrivateKeyRoute = (props: RouteProps) => {
 const HomeRoute = () => {
   const applicationState = useSelector(applicationStateSelector);
   const publicKey = useSelector(publicKeySelector);
+  const error = useSelector(authErrorSelector);
+  if (applicationState === APPLICATION_STATE.APPLICATION_ERROR) {
+    return <AppError>{error}</AppError>;
+  }
```

Comment:
This feels like there's an opportunity for abstraction, but I didn't dive into it. 3x copy-pasted data loading + early returns